src/query/functions/temporal/base.go

```diff
@@ -184,27 +164,41 @@ type blockMeta struct {
 	end         xtime.UnixNano
 	aggDuration xtime.UnixNano
 	stepSize    xtime.UnixNano
+	queryCtx    *models.QueryContext
 	steps       int
-	seriesMeta  []block.SeriesMeta
 }
 
-func batchProcess(
+func (c *baseNode) batchProcess(
 	ctx context.Context,
+	b block.Block,
 	iterBatches []block.SeriesIterBatch,
-	builder block.Builder,
 	m blockMeta,
-	p processor,
-) error {
+) (block.Builder, error) {
 	var (
-		metas = m.seriesMeta
-
 		mu       sync.Mutex
 		wg       sync.WaitGroup
 		multiErr xerrors.MultiError
 		idx      int
 	)
 
-	builder.PopulateColumns(len(metas))
+	meta := b.Meta()
+	builder, err := c.controller.BlockBuilder(m.queryCtx, meta, nil)
```

Comment:
Ah, was sharing the builder concurrently causing issues?

Reply:
Yeah finally just got the other issue where it was sharing the processor, about to put up a PR after I manage to get a good repro case for it 👍